Makefile

```diff
@@ -1,64 +1,63 @@
 	@echo
 	@echo usage:
 	@echo
 	@echo "  make install         -- install dev setup"
-	@echo "  make lint            -- lint all project files"
-	@echo "  make lint-quick      -- lint all files staged in git"
+	@echo "  make clean           -- delete node modules and venv"
 	@echo "  make server          -- start a dev server"
 	@echo "  make watch           -- start a dev server and rebuild js and css files on changes"
+	@echo "  make background      -- start a dev server, rebuild js and css files on changes, and start background processes"
 	@echo "  make test            -- run all test cases with pytest"
-	@echo "  make makemessages    -- create new po files from the source"
-	@echo "  make compilemessages -- create new mo files from the translated po files"
+	@echo "  make test-lastfailed -- run test that failed last"
+	@echo "  make test-clean      -- test on new database"
+	@echo "  make coverage        -- write coverage report to dir htmlcov"
+	@echo "  make lint            -- lint all project files"
+	@echo "  make lint-quick      -- lint all files staged in git"
+	@echo "  make po              -- create new po files from the source"
+	@echo "  make mo              -- create new mo files from the translated po files"
 	@echo "  make release         -- build everything required for a release"
 	@echo
 
 .PHONY: install
 install:
 	npm install --no-save
+	npm run build
 	if [ ! -f $(VIRTUAL_ENV)/bin/python3 ]; then python3 -m venv $(VIRTUAL_ENV); fi
-	$(VIRTUAL_ENV)/bin/python3 -m pip install -r requirements/dev.txt
+	$(VIRTUAL_ENV)/bin/python3 -m pip install --upgrade -r requirements/dev.txt
 	$(VIRTUAL_ENV)/bin/python3 manage.py migrate
 
-.PHONY: makemessages
-makemessages:
-	$(VIRTUAL_ENV)/bin/python manage.py makemessages -d django
-	$(VIRTUAL_ENV)/bin/python manage.py makemessages -d djangojs
-	sed -i 's%#: .*/adhocracy4%#: adhocracy4%' locale/*/LC_MESSAGES/django*.po
-	sed -i 's%#: .*/a4-meinberlin%#: a4-meinberlin%' locale/*/LC_MESSAGES/django*.po
-	msgen locale/en_GB/LC_MESSAGES/django.po -o locale/en_GB/LC_MESSAGES/django.po
-	msgen locale/en_GB/LC_MESSAGES/djangojs.po -o locale/en_GB/LC_MESSAGES/djangojs.po
-
-.PHONY: compilemessages
-compilemessages:
-	$(VIRTUAL_ENV)/bin/python manage.py compilemessages
+.PHONY: clean
+clean:
+	if [ -d node_modules ]; then rm -r node_modules; fi
+	if [ -d venv ]; then rm -r venv; fi
 
 .PHONY: server
 server:
 	$(VIRTUAL_ENV)/bin/python3 manage.py runserver 8004
 
 .PHONY: watch
 watch:
+	trap 'kill %1' KILL; \
 	npm run watch & \
 	$(VIRTUAL_ENV)/bin/python3 manage.py runserver 8004
 
-.PHONY: lint
-lint:
-	. $(VIRTUAL_ENV)/bin/activate && $(NODE_BIN)/polylint
-
-.PHONY: lint-quick
-lint-quick:
-	. $(VIRTUAL_ENV)/bin/activate && $(NODE_BIN)/polylint -SF
+.PHONY: background
+background:
+	trap 'kill %1; kill %2' KILL; \
+	npm run watch & \
+	$(VIRTUAL_ENV)/bin/python3 manage.py process_tasks & \
+	$(VIRTUAL_ENV)/bin/python3 manage.py runserver 8004
```

Comment:
Add the process_tasks to watch instead of having a dedicated command. Any trade offs with this approach?

Reply:
Sometimes Ctrl-C somehow does not stop the processes (esp. on macs). So haveing a make watch without that makes sense.

Reply:
yeah, `&` is not a process management. I thouhg the trap thing solves exactly hiting the Ctrl-C thing. I don't like this workaround but ok, leave it there.